src/commands/edge_app.rs

```diff
@@ -0,0 +1,250 @@
+use crate::authentication::Authentication;
+use crate::commands::CommandError;
+use std::collections::HashMap;
+
+use log::debug;
+use reqwest::header::HeaderMap;
+use reqwest::StatusCode;
+use serde::{Deserialize, Serialize};
+use serde_yaml;
+use std::fs;
+use std::fs::File;
+use std::io::Write;
+use std::path::Path;
+
+#[derive(Clone, Debug, Default, PartialEq, Serialize, Deserialize)]
+pub struct EdgeAppManifest {
+    #[serde(default)]
+    pub id: String,
+    #[serde(default)]
+    pub created_at: String,
+    #[serde(default)]
+    pub created_by: String,
+    #[serde(default)]
+    pub permissions: Vec<String>,
+    #[serde(default)]
+    pub updated_at: String,
+    pub name: String,
+    pub version: String,
+    pub description: String,
+    pub icon: String,
+    pub author: String,
+    pub homepage_url: String,
+}
+
+pub struct EdgeAppCommand {
+    authentication: Authentication,
+}
+
+impl EdgeAppCommand {
+    pub fn new(authentication: Authentication) -> Self {
+        Self { authentication }
+    }
+
+    pub fn init(self, path: &Path) -> Result<(), CommandError> {
+        let mut object = serde_yaml::to_value(EdgeAppManifest::default())?;
+        let map = object.as_mapping_mut().ok_or(CommandError::MissingField)?;
+
+        // following fields will be generated server side when publishing.
+        map.remove("id");
+        map.remove("created_at");
+        map.remove("created_by");
+        map.remove("updated_at");
```

Comment:
It seems you don't use these fields at all: created_at, created_by and updated_at both at init and on publish - may be just remove them from json struct - and they I guess should be ignored when parsing.

Reply:
no - they won't be ignored, parsing will faile. Though it might configurable. 
I am using them when publishing. At that point, they go to the ready file. If we don't need them I think we should remove them from the table at which point I can modify the application.

Reply:
We do need them at the db and I guess it could be usable to API user as information - but is not manifest data. I believe it could be configurable though.